server/site.go

```
@@ -371,24 +373,93 @@ func (s *Site) WriteFrontMatter(identifier wikipage.PageIdentifier, fm wikipage.
 	return p.Update(newText)
 }
 
-func lenientParse(content []byte, matter any) (body []byte, err error) {
-	body, err = adrgfrontmatter.Parse(bytes.NewReader(content), matter)
+func (s *Site) lenientParse(content []byte, matter any, identifier wikipage.PageIdentifier) (body []byte, err error) {
+	// Apply frontmatter migrations BEFORE parsing
+	migratedContent := content
+	migrationApplied := false
+	
+	if s.MigrationApplicator != nil {
```

no silent errors. this should barf if its nil